test/index.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+/* eslint-env mocha */
+/* eslint no-proto: 0 */
+var assert = require('assert')
+var setPrototypeOf = require('..')
+
+function protoTest (obj, proto) {
+  if (Object.getPrototypeOf) {
+    assert.strictEqual(Object.getPrototypeOf(obj), proto)
+  } else if ({ __proto__: [] } instanceof Array) {
+    assert.strictEqual(obj.__proto__, proto)
+  } else {
+    for (var prop in obj) {
```

Comment:
I think that it is best to just loop over prop not obj because you are checking to see if all the fields that are in props and not found on obj was added to obj.